providers/netty4/src/main/java/org/asynchttpclient/netty/handler/Processor.java

```diff
@@ -71,11 +75,41 @@ public void channelRead(final ChannelHandlerContext ctx, Object msg) throws Exce
                 ac.call();
                 Channels.setDiscard(channel);
             }
+            ReferenceCountUtil.release(msg);
+
 
         } else if (attribute instanceof NettyResponseFuture) {
             NettyResponseFuture<?> future = (NettyResponseFuture<?>) attribute;
             protocol.handle(channel, future, msg);
 
+        } else if (attribute instanceof StreamedResponsePublisher) {
```

Comment:
Actually, there's 2 cases where retry makes sense. Sadly, AHC doesn't currently tell them appart, as it was originally intended to deal with file downloading.

1. The request was sent, and the connection crashed while the response body chunks were being received.
2. The request was not sent, because the connection crashed first.

The second case typically happens when reusing a pooled connection, and the remote peer closes it (keep-alive idle timeout, server being overloaded). We do check that the connection is alive when pooling it, and just before writing, but there's always a time window where this could happen. Under heavy load, this happens all the time, as soon as the server implements idle timeout. Browsers do implement a retry to deal with such use case. AHC has a maxRetry default value of 4.

Note that you can be notified of a retry when you implement `AsyncHandlerExtensions`, such you can clean/reset chunks that you've received so far.

What's the expected behavior of reactive-streams when the connection crashes? Is it up to the user to implement retry logic?